service/src/main/java/uk/gov/hmcts/reform/fpl/utils/SecureDocStoreHelper.java

```diff
@@ -0,0 +1,189 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.reform.ccd.document.am.model.Classification;
+import uk.gov.hmcts.reform.ccd.document.am.model.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.service.FeatureToggleService;
+import uk.gov.hmcts.reform.fpl.service.SecureDocStoreService;
+
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+
+@Slf4j
+public class SecureDocStoreHelper {
+
+    private FeatureToggleService featureToggleService;
+    private SecureDocStoreService secureDocStoreService;
+
+    public SecureDocStoreHelper(SecureDocStoreService secureDocStoreService,
+                                FeatureToggleService featureToggleService) {
+        this.featureToggleService = featureToggleService;
+        this.secureDocStoreService = secureDocStoreService;
+    }
+
+    public byte[] download(final String documentUrlString) {
+        return download(documentUrlString, null);
+    }
+
+    /**
+     * If secure doc store toggle is off, attempt to fire the API and log any exceptions caught.
+     * @param documentUrlString  document url in string
+     * @param oldDmStoreApproach if featureToggleService.isSecureDocstoreEnabled() = false, it is a mandatory parameter.
+     *                           otherwise, UnsupportedOperationException will be thrown.
+     * @return byte array of the file
+     */
+    @SneakyThrows
+    public byte[] download(final String documentUrlString, Callable<byte[]> oldDmStoreApproach) {
+        try {
+            log.info("Downloading document: {}", documentUrlString);
+            byte[] bytesFromSecureDocStore = secureDocStoreService.downloadDocument(documentUrlString);
+            if (featureToggleService.isSecureDocstoreEnabled()) {
+                return bytesFromSecureDocStore;
+            } else {
+                log.info("Downloaded document attempted from CDAM without error: {}", documentUrlString);
+            }
+        } catch (Exception t) {
+            if (!featureToggleService.isSecureDocstoreEnabled()) {
+                log.error("↑ ↑ ↑ ↑ ↑ ↑ ↑ EXCEPTION CAUGHT (SECURE DOC STORE: DISABLED) ↑ ↑ ↑ ↑ ↑ ↑ ↑", t);
+            } else if (oldDmStoreApproach == null) {
+                throw t;
+            }
+        }
+        if (!featureToggleService.isSecureDocstoreEnabled() && !isEmpty(oldDmStoreApproach)) {
+            log.info("Using old dm-store approach to download the document: {}.", documentUrlString);
```

Comment:
Same as this one, I think we can skip this log before we deploy, to save logging space